### PR TITLE
[LMS-334] Notify about missing descriptors for Assignment Category and Submission Status

### PR DESCRIPTION
**Pull request by lalrg**

### CODE REVIEW

**Comment by stephenfuqua** [comment]
@lalrg I uploaded the original sample data files, because I knew that they would have bad descriptors. I got the following warning message:

> 2021-08-10 20:51:26,454 - WARNING - edfi_lms_harmonizer.exceptions_reports - There are 4 missing descriptors for Assignment Category and 9 missing descriptors for Submission Status

However, the exception file only lists two assignment categories and four submission statuses. Should these two counts match?

File: src/lms-harmonizer/edfi_lms_harmonizer/exceptions_reports.py [hunk 1]
```diff
@@ -68,3 +99,15 @@ def create_exception_reports(adapter: Adapter, output_directory: str) -> None:
...
+        "Writing the Assignment Category and Submission Status missing descriptors report"
+    )
+    assignment_cat = pd.read_sql(QUERY_FOR_ASSIGNMENT_CAT_DESCRIPTORS, adapter.engine)
+    submission_status = pd.read_sql(
+        QUERY_FOR_SUBMISSION_STATUS_DESCRIPTORS, adapter.engine
+    )
+    descriptors_report = pd.concat([assignment_cat, submission_status], axis=1)
+    descriptors_report.to_csv(
+        _get_file_path(output_directory, DESCRIPTORS), index=False
+    )
```
Comment (stephenfuqua) [inline src/lms-harmonizer/edfi_lms_harmonizer/exceptions_reports.py:113]: Here's a sample output right now:

```
AssignmentCategory,SubmissionStatus
Assessment,Created
Assignment,Late
,on-time
,Returned
```

Translated to a markdown table for reading:

|AssignmentCategory | SubmissionStatus |
| -- |-- |
|Assessment|Created|
|Assignment|Late|
||on-time|
||Returned|

Reading it in Excel, like the markdown above, is pretty straightforward. But it does look weird in CSV.

What do you think about changing to the following output format? Also note the addition of the expected namespace

```
Descriptor, Namespace, Missing Value
AssignmentCategoryDescriptor, uri://ed-fi.org/edfilms/AssignmentCategoryDescriptor/BestLMS, Assessment
AssignmentCategoryDescriptor, uri://ed-fi.org/edfilms/AssignmentCategoryDescriptor/BestLMS, Assignment
SubmissionStatusDescriptor, uri://ed-fi.org/edfilms/SubmissionStatusDescriptor/BestLMS, Created
SubmissionStatusDescriptor, uri://ed-fi.org/edfilms/SubmissionStatusDescriptor/BestLMS, Late
SubmissionStatusDescriptor, uri://ed-fi.org/edfilms/SubmissionStatusDescriptor/BestLMS, on-time
SubmissionStatusDescriptor, uri://ed-fi.org/edfilms/SubmissionStatusDescriptor/BestLMS, Returned
```

File: extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1110-view-missing-assignmentsubmission-status-descriptors.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+CREATE VIEW lmsx.missing_assignment_submission_status_descriptors AS
+    SELECT * FROM LMS.AssignmentSubmission lmsAssignmentSubmission
+    WHERE NOT EXISTS(
+            SELECT 1 FROM edfi.Descriptor assignmentCatDescriptor WHERE
+                assignmentCatDescriptor.CodeValue = lmsAssignmentSubmission.SubmissionStatus
+                AND assignmentCatDescriptor.Namespace = 'uri://ed-fi.org/edfilms/AssignmentCategoryDescriptor/' + lmsAssignmentSubmission.SourceSystem
```
Comment (stephenfuqua) [inline extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1110-view-missing-assignmentsubmission-status-descriptors.sql:11]:
```suggestion
                AND assignmentCatDescriptor.Namespace = 'uri://ed-fi.org/edfilms/SubmissionStatusDescriptor/' + lmsAssignmentSubmission.SourceSystem
```

**Review by stephenfuqua — CHANGES_REQUESTED** [review]

File: src/lms-harmonizer/edfi_lms_harmonizer/exceptions_reports.py (hunk 1 above)
Comment (lalrg) [reply src/lms-harmonizer/edfi_lms_harmonizer/exceptions_reports.py:113]: Yes, will commit a new version in a few minutes

File: extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1100-view-missing-assignment-category-descriptors.sql [hunk 3]
```diff
@@ -0,0 +1,12 @@
+-- SPDX-License-Identifier: Apache-2.0
```
Comment (stephenfuqua) [inline extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1100-view-missing-assignment-category-descriptors.sql:1]: Should this file move to the new `scripts` directory under the Harmonizer src?

Comment (stephenfuqua) [reply extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1100-view-missing-assignment-category-descriptors.sql:1]: Also you'll need to enter these two script names into the new migrator module.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**